lib/build-components/build_components/__main__.py

```diff
@@ -0,0 +1,511 @@
+            outfile.write(
+                textwrap.dedent(
+                    """
+                    # This is a dynamically generated file created by build-components for the
+                    # purpose of building the component containers.
+                    # It is copied into our docker images to globally configure poetry
+
+                    [virtualenvs]
+                    # We're in a docker container, there's no need for virtualenvs
+                    # One should still configure pip to use "--user" behavior so that
+                    # poetry-installed packages will be placed in ~/.local
+                    create = false
+                    """
+                )
+            )
+
+        with open(dockerfile_path, "w") as outfile:
+            outfile.write(dockerfile)
+
+        yield dockerfile_path
+    finally:
+        with contextlib.suppress():
+            dockerfile_path.unlink()
+
+        with contextlib.suppress():
+            pip_conf_path.unlink()
+
+        with contextlib.suppress():
+            poetry_toml_path.unlink()
+
+
+def build_editor_image(build_info: BuildInfo) -> None:
+    """
+    Build a companion image to the final built image that removes the ENTRYPOINT and CMD settings.
+
+    This image can be used for shelling into for debugging and/or running arbitrary commands in a
+    mirror of the built image.
+
+    The resulting image will be tagged as ``{project}-{component}:editor``
+
+    :param build_info: The build info populated from the config and command line arguments.
+    """
+    logger.info("Building editor image for %s component", build_info.component)
+
+    # Perform a "no context" docker build
+    _docker_build(
+        tags=build_info.editor_tag,
+        dockerfile=textwrap.dedent(
+            f"""
+            FROM {build_info.tag}
+            CMD "/bin/sh"
+            ENTRYPOINT []
+            """
+        ).encode(),
+    )
+
+
+def _docker_build(
+    tags: typing.Union[str, typing.List[str]],
+    buildargs: dict = None,
+    dockerfile: typing.Union[pathlib.Path, bytes] = None,
+) -> None:
+    """
+    A convenience wrapper for calling out to "docker build".
+
+    We always send the same context: the entire components/ directory.
+
+    :param tags: The tags to be applied to the built image.
+    :param buildargs: Values for ARG instructions in the dockerfile.
+    :param dockerfile: The dockerfile to build against. If not provided, it's assumed that a
+                       Dockerfile is present in the context directory. If it's a bytes object, it
+                       will be provided to the docker build process on stdin and a "no context"
+                       build will take place. Otherwise, a normal docker build will be performed
+                       with the specified Dockerfile.
+    """
+    buildargs = buildargs or {}
+    buildargs.setdefault("CACHE_BUST", str(time.time()))
+
+    cmd = ["env", "DOCKER_BUILDKIT=1", "docker", "build"]
```

Comment:
If we enable DOCKER_BUILDKIT, it will cause failures in Jenkins build `buildkit not supported by daemon`

Reply:
Updating our jenkins instance to version that supports it is not an option?

Reply:
Don't know if we can upgrade the jenkins before this can land. @jarojasm95 - your thoughts?
It seems better to not use BUILDKIT in first pass, upgrade Jenkins (in future) and bring in BUILDKIT then.
That way this PR won't get blocked on Jenkins upgrade.

Reply:
how about if i make it configurable? defaulted to on, but you can disable?

Reply:
ok. Configurable option works!

Reply:
```